src/Microsoft.IdentityModel.Logging/LogHelper.cs

```diff
@@ -375,20 +373,22 @@ private static object SanitizeSecurityArtifact(object arg)
             return arg;
         }
 
-        private static string? RemovePII(object arg)
+        private static string RemovePII(object arg)
         {
             if (arg is Exception ex && IsCustomException(ex))
                 return ex.ToString();
 
             if (arg is NonPII)
-                return arg!.ToString();
+                return arg.ToString()!; // the null check is done in FormatInvariant
```

Comment:
I find those situations tricky. Although it is flow correct it might become fragile for change during refactoring as moving it around might change the dependency, but good one.